### ISSUE TITLE
Display OSLog events to dap-ui

### ISSUE THREAD

**Issue by Almaz5200** [wojciech-kulik/xcodebuild.nvim#13]
I think I've found a way to get logs from Simulator outside of Xcode: there is a terminal command
`xcrun simctl spawn <SIM_UUID> log stream --predicate 'process == "<Process_Name>"'`

It will stream all logs from the app if ran in terminal.
Also you can set up some filter:
`xcrun simctl spawn <SIM_UUID> log stream --predicate 'process == "<Process_Name>" AND subsystem == "app"'`
In order to filter out irrelevant logs.

I'd be happy to contribute and implement it myself, but I am unsure where to start. I'd be really grateful if you could point me in the right direction on where to start and I'll happily implement it.
Unfortunately this does not cover print statements, but OSLog events are a great improvement already

**Comment by wojciech-kulik** [comment]
Hi,

`nvim-dap` has a window called `console`. I'm not sure how it works, but maybe it would be possible to integrate with that and show logs there: https://github.com/mfussenegger/nvim-dap/blob/master/doc/dap.txt (check `dap-terminal`)

If so, you could just add a function into `dap.lua` which users could use to manage that.

All xcode commands are in `xcode.lua` when you launch the the `launch_app` function is called. You could extend it to keep logging. But keep in mind that we want to see logs only when we are going to debug the app. Fortunatel, this is passed as a parameter to the function.

Sample function to launch the app and print logs:
```
xcrun simctl launch --console <SIMID> <bundle id> log stream
```

**Comment by Almaz5200** [comment]
<img width="699" alt="Screenshot 2023-12-08 at 19 32 42" src="https://github.com/wojciech-kulik/xcodebuild.nvim/assets/21291840/33dd8f24-ea40-4e6e-8b45-c446504c1d86">

For now I wasn't able to put anything in console, but I was able to put logs to REPL buffer like that
I did it by running logs stream alongside codelldb which dap interprets as server stdout

```
#!/bin/bash
"$HOME/nvimPlugins/codelldb/extension/adapter/codelldb" \
    --port 13000 \
    --liblldb "/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Versions/A/LLDB" &
xcrun simctl spawn CE53EEA4-D829-4918-BA2E-0105873576F6 log stream --predicate 'process == "TribunaUA" AND subsystem == "app"' | cut -d ' ' -f35- &
wait
```
```
dap.adapters.codelldb = {
  type = "server",
  port = "13000",
  executable = {
    command = os.getenv("HOME") .. "/.config/nvim/lua/custom/configs/runDap.sh",
  },
}
```
Unfortunately it presents logs with severe delay, but I will continue to explore it

**Comment by wojciech-kulik** [comment]
Right, it makes sense that REPL can see the output of the process that it's attached to 👍 .

**Comment by Almaz5200** [comment]
Hey, so unfortunately I wasn't able to either put logs to the console or get rid of that huge delay in logs. But I use tmux, so in the end, I found it easier to just open another terminal window below my nvim window and stream logs there, which looks like this. I've also made a zsh func for it so that it's quicker to start

<img width="1728" alt="Screenshot 2023-12-11 at 13 11 13" src="https://github.com/wojciech-kulik/xcodebuild.nvim/assets/21291840/8d235495-c4b7-4567-afe2-b585821514a6">

This also has added benefit of preserving color coding which goes a long way in terms of making it simpler to navigate visually

**Comment by wojciech-kulik** [comment]
Looks good, I'm also using tmux, I will check out this solution. What command do you use to stream logs?

**Comment by Almaz5200** [comment]
`xcrun simctl spawn CE53EEA4-D829-4918-BA2E-0105873576F6 log stream --predicate 'process == "TribunaUA" AND subsystem == "app"'`

Although I think it would be a good idea to get sim id and scheme from settings.json since it's already there, I'll try to put together a script for that

**Comment by Almaz5200** [comment]
`subsystem == "app"` is not a system thing also, that's just how we define all our loggers

```
extension Logger {

    public static let debugger: Logger = Logger(subsystem: "app", category: "debuger")
    public static let data: Logger = Logger(subsystem: "app", category: "data")
    public static let interface: Logger = Logger(subsystem: "app", category: "interface")
    public static let ads: Logger = Logger(subsystem: "app", category: "ads")
    public static let subscription: Logger = Logger(subsystem: "app", category: "subscription")

}
```

**Comment by Almaz5200** [comment]
```
#!/bin/zsh

settings_file=".nvim/xcodebuild/settings.json"

dest=$(python -c "import json; print(json.load(open('$settings_file'))['destination'])")
scheme=$(python -c "import json; print(json.load(open('$settings_file'))['scheme'])")

echo "Streaming logs from $scheme at $dest"

xcrun simctl spawn $dest log stream --predicate "process == \"$scheme\" AND subsystem == \"app\""
```

Here, that parses settings.json and fills in scheme and simulator id.
I am not sure there's a way to distribute that solution with the plugin though, only probably by putting this script in user's project folder automatically, but that seems a bit invasive 😅

**Comment by Almaz5200** [comment]
Sorry, closed accidentally!
If you like the idea of putting a script like that in user's project folder with ability to configure that behaviour in plugin setup I can implement that

**Comment by wojciech-kulik** [comment]
I think for now it will be better to keep it outside of the plugin. I will convert it into discussion to keep some ideas there
